### PR TITLE
umdns: fix start script when 'seccomp' is not installed

### PR DESCRIPTION
**Pull request by lars18th**
The umdns service can actually operate with and without seccomp enabled.
However, now when seccomp is not installed, the service script cannot start 
the umdns service because the seccomp binary is missing.

In this case the System Log prints out a bunch of lines similar to:
```daemon.err dawn[2194]: Failed to look up test object for umdns```

This patch fixes the problem by checking both the presence of the 
configuration file and the binary file of the seccomp package.

Signed-off-by: Lars The mail@lars18th.net

### CODE REVIEW

**Comment by adschm** [comment]
This has formal issues:

 * Commit title needs to be improved
 * Commit message needs to be wrapped
 * Signed-off-by is missing

https://openwrt.org/submitting-patches
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes

**Comment by adschm** [comment]
... and "Chech" -> "check"

**Comment by adschm** [comment]
Hmm, you probably didn't read "Frequent PR mistakes" ...

**Comment by wulfy23** [comment]
is `/etc/seccomp/umdns.json` not provided by seccomp?

**Comment by wulfy23** [comment]
ok... so this file is provided by umdns...

was asking where it came from

**Comment by kirelagin** [comment]
There is already a ticket for this: https://bugs.openwrt.org/index.php?do=details&task_id=3355, there is analysis in there.

The script does not try to call the `seccomp` binary, what fails is `procd` trying to call `/sbin/seccomp-trace`, so I am not sure the fix proposed in this PR is the right one. This seems like a deeper problem.

**Comment by lars18th** [comment]
Hi @kirelagin ,

> There is already a ticket for this: https://bugs.openwrt.org/index.php?do=details&task_id=3355, there is analysis in there.
> 
> The script does not try to call the `seccomp` binary, what fails is `procd` trying to call `/sbin/seccomp-trace`, so I am not sure the fix proposed in this PR is the right one. This seems like a deeper problem.

Thank you, I readed the two tickets:
- https://bugs.openwrt.org/index.php?do=details&task_id=3355
- https://bugs.openwrt.org/index.php?do=details&task_id=4138

But I feel this is not the **root cause** of the problem fixed with this patch: The fact is what I print in the fist line of my description:

> _The umdns service can actually operate with and without seccomp enabled._

For a reason that I don't understand, the writer of the original script has the assumption that the umdns service requires the seccomp binary. The use of policies and jails it's _optional_ to run this service. In fact, I'm running it in this way with this patch. So, in order to fix _this_ problem the script requires to be improved. Another question is to fix the problem of the procd dependency with the seccomp.

In other words: If a user wants to use policies and jails, then s/he will wants to install the related packages. In this case, the running services will require to set the policies. But if the user doesn't want to use them (because the overhead), then all processes requires to run without them. That's the question.

So the _current_ script has the error to think that policies are in use. And the reasonable solution is to check this before call to seccomp.
Don't you think so?

**Comment by lars18th** [comment]
Hi @kirelagin ,

> The script does not try to call the `seccomp` binary, what fails is `procd` trying to call `/sbin/seccomp-trace`, so I am not sure the fix proposed in this PR is the right one. This seems like a deeper problem.

Sure? What actually does the script now is this:
```
	[ -f /etc/seccomp/umdns.json ] && procd_set_param seccomp /etc/seccomp/umdns.json
```
and here the patched version:
```
	[ -f /etc/seccomp/umdns.json -a -f /sbin/seccomp ] && procd_set_param seccomp /etc/seccomp/umdns.json
```

So the current implementation instructs the procd to call to `seccomp` even if this binary is not present in the system. And that's a mistake.

Take note that this patch is running on several devices for some months without any error. So the patch actually works. And I don't see any reason to not merge it.

Regards.

**Comment by kirelagin** [comment]
First of all, `/sbin/seccomp` is not a thing.

**Comment by lars18th** [comment]
> First of all, `/sbin/seccomp` is not a thing.

OK. Now I understand your response. Thank you!
In my case it always runs because the test of `/sbin/seccomp` is false.

So then the solution is to check -inside the script- if the policies and jails are running.
You've any idea to achieve it?
